src/autoscaler/metricscollector/cmd/metricscollector/main.go

```diff
@@ -107,33 +107,40 @@ func main() {
 		return nil
 	})
 
-	httpServer, err := server.NewServer(logger, conf, cfClient, noaa, instanceMetricsDB)
+	httpServer, err := server.NewServer(logger.Session("http_sever"), conf, cfClient, noaa, instanceMetricsDB)
 	if err != nil {
 		logger.Error("failed to create http server", err)
 		os.Exit(1)
 	}
 
-	consulClient, err := consuladapter.NewClientFromUrl(conf.Lock.ConsulClusterConfig)
-	if err != nil {
-		logger.Fatal("new consul client failed", err)
-	}
-
-	serviceClient := metricscollector.NewServiceClient(consulClient, mcClock)
-
-	lockMaintainer := serviceClient.NewMetricsCollectorLockRunner(
-		logger,
-		generateGUID(logger),
-		conf.Lock.LockRetryInterval,
-		conf.Lock.LockTTL,
-	)
-
-	registrationRunner := initializeRegistrationRunner(logger, consulClient, conf.Server.Port, mcClock)
-
-	members := grouper.Members{
-		{"lock-maintainer", lockMaintainer},
-		{"collector", collectServer},
-		{"http_server", httpServer},
-		{"registration", registrationRunner},
+	var members grouper.Members
+
+	if conf.Lock.ConsulClusterConfig == "" {
+		members = grouper.Members{
+			{"collector", collectServer},
+			{"http_server", httpServer},
+		}
+	} else {
+		consulClient, err := consuladapter.NewClientFromUrl(conf.Lock.ConsulClusterConfig)
+		if err != nil {
+			logger.Fatal("new consul client failed", err)
+		}
+
+		serviceClient := metricscollector.NewServiceClient(consulClient, mcClock)
+		lockMaintainer := serviceClient.NewMetricsCollectorLockRunner(
+			logger,
+			generateGUID(logger),
+			conf.Lock.LockRetryInterval,
+			conf.Lock.LockTTL,
+		)
+
+		registrationRunner := initializeRegistrationRunner(logger, consulClient, conf.Server.Port, mcClock)
+		members = grouper.Members{
```

Comment:
It would be better to add the common runner i.e http_server and collector once and prepend lock_maintainer and apend registrartion runner for consul